Sources/Resource/RetryableResource.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+
+/// A type representing a resource that can be retried after failing.
+public protocol RetryableResource {
+
+    /// A type that represents a received remote payload.
+    associatedtype Remote
+
+    /// A type that represents a network request.
+    associatedtype Request
+
+    /// A type that represents a network response.
+    associatedtype Response
+
+    /// The resource's specialized retry policy.
+    typealias RetryPolicy = ResourceRetry.Policy<Remote, Request, Response>
+
+    /// The errors that have resulted in a retry after having occurred.
+    var retriedAfterErrors: [Error] { get set }
```

Comment:
It might be just me, but I find this name hard to interpret. Would `pastErrors` be enough/ better?

Reply:
We're using `previousErrors` on other API's (i.e. `ResourceRetry.Policy`), but I went with `retriedAfterErrors` on the Resource to make it clear that they are errors related with the retry "behavior". They're in also the errors that have in fact resulted in a retry after having occurred 😝.

However it the name is confusing or hard to interpret, I guess we can change it.